modules/api-import-export/src/main/java/org/wso2/carbon/apimgt/importexport/APIService.java

```diff
@@ -260,4 +262,89 @@ public Response importAPI(@Multipart("file") InputStream uploadedInputStream, @Q
             }
         }
     }
+
+    /**
+     * This service will update an existing API given by the ID. It will preserve state of the API during the update
+     *
+     * @param apiID       ID for the API
+     * @param httpHeaders HTTP headers for authentication mechanism
+     * @return A JSON response indicating the status of the update
+     */
+    @PUT
+    @Path("/{apiID}")
+    @Consumes(MediaType.MULTIPART_FORM_DATA)
+    @Produces(MediaType.APPLICATION_JSON)
+    public Response updateAPI(@Multipart("file") InputStream uploadedInputStream, @PathParam("apiID") String apiID,
+                              @QueryParam("preserveProvider") String defaultProviderStatus, @Context HttpHeaders httpHeaders) {
+        boolean isTenantFlowStarted = false;
+        boolean isProviderPreserved = true;
+
+        //Check if the URL parameter value is specified, otherwise the default value "true" is used
+        if (APIImportExportConstants.STATUS_FALSE.equalsIgnoreCase(defaultProviderStatus)) {
+            isProviderPreserved = false;
+        }
+
+        try {
+            Response authorizationResponse = AuthenticatorUtil.authorizeUser(httpHeaders);
+
+            if (Response.Status.OK.getStatusCode() != authorizationResponse.getStatus()) {
+                return authorizationResponse;
+            }
+
+            AuthenticationContext authenticationContext = (AuthenticationContext) authorizationResponse.getEntity();
+            String tenantDomain = MultitenantUtils.getTenantDomain(authenticationContext.getUsername());
+            String currentUser = authenticationContext.getDomainAwareUsername();
+
+            if (!tenantDomain.equals(MultitenantConstants.SUPER_TENANT_DOMAIN_NAME)) {
```

Comment:
Possible NPE here. Lets change this as following.

if (!MultitenantConstants.SUPER_TENANT_DOMAIN_NAME.equals(tenantDomain)) {

Reply:
sure, will do